ingestion-edge/ingestion_edge/disk_queue.py

```diff
@@ -0,0 +1,70 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, you can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .conf import QUEUE_FILE
+from .conf import QUEUE_POSITION_FILE
+from flask import Blueprint
+from google.cloud.pubsub import PublisherClient
+from os import remove
+from os.path import exists
+from threading import Lock
+from typing import Tuple
+import ujson
+
+disk_queue = Blueprint("disk_queue", __name__)
+client = PublisherClient()
+read_lock = Lock()
+write_lock = Lock()
+
+
+def write(**kwargs):
+    with write_lock:
+        # TODO disable autoscaling
+        with open(QUEUE_FILE, "a") as q:
+            q.write(ujson.dumps(kwargs) + "\n")
```

Comment:
I'm not sure about `ujson`, but the standard library `json` has both `json.dumps` and `json.dump`, where `dumps` is "**dump** to **s**tring". If `ujson` works the same, I expect this would be better as `ujson.dump(kwargs, q)`. That wouldn't automatically include the newline at the end, but I think an additional `q.write("\n")` would be fine.

Mostly I expect this to be more efficient than generating an intermediate string and then writing it to a file, but I've never profiled it.